### PR TITLE
Adding output vars to JenkinsQueueJob task

### PR DESCRIPTION
**Pull request by jpricketMSFT**
- Since output vars require a new version of the agent, I bumped the major version of the task to 2 (this allows the user to choose when to upgrade)
- This is part of internal user story #982063

### CODE REVIEW

File: Tasks/JenkinsQueueJob/jenkinsqueuejobtask.ts [hunk 1]
```diff
@@ -111,6 +111,9 @@ async function doWork() {
         var rootJob = await util.pollCreateRootJob(queueUri, jobQueue, taskOptions);
         //start the job queue
         jobQueue.start();
+        //store the job name in the output variable
+        tl.debug('JENKINS_JOB_ID=' + rootJob.executableNumber);
+        tl.setVariable("JENKINS_JOB_ID", rootJob.executableNumber.toString());
```
Comment (DavidStaheli) [inline Tasks/JenkinsQueueJob/jenkinsqueuejobtask.ts]: I need to double-check, but I think calling `setVariable()` already provides debug output.

File: Tasks/JenkinsQueueJob/task.json [hunk 2]
```diff
@@ -99,6 +99,12 @@
             }
         }
     ],
+    "outputVariables" : [
+        {
+            "name" : "JENKINS_JOB_ID",
+            "description" : "The id of the Jenkins Job instance queued by this task. Use this variable in the Jenkins Download task to download the artifacts for this particular job instance."
```
Comment (DavidStaheli) [inline Tasks/JenkinsQueueJob/task.json]: ID should be capitalized.  Job should be lowercase.  "Jenkins Download task" should be "Jenkins Download Artifacts task".

File: Tasks/JenkinsQueueJob/jenkinsqueuejobtask.ts (hunk 1 above)
Comment (DavidStaheli) [inline Tasks/JenkinsQueueJob/jenkinsqueuejobtask.ts]: Yes, see how the task lib already outputs debug info [here](https://github.com/Microsoft/vsts-task-lib/blob/212473e933d4d56416d1d1c91280d88c0c75c573/node/task.ts#L120).

Comment (jpricketMSFT) [inline Tasks/JenkinsQueueJob/jenkinsqueuejobtask.ts]: Good point. I will remove my extra logging.

File: Tasks/JenkinsQueueJob/task.json (hunk 2 above)
Comment (jpricketMSFT) [inline Tasks/JenkinsQueueJob/task.json]: Thanks, I will correct these words.

**Pull request merged**